src/ashell/comms-uart.c

```diff
@@ -322,31 +303,18 @@ void comms_write_buf(const char *buf, int len)
     if (len == 0)
         return;
 
-    struct device *dev = dev_upload;
-    uart_irq_tx_enable(dev);
-
+    uart_irq_tx_enable(dev_upload);
     data_transmitted = false;
-    uart_fifo_fill(dev, buf, len);
-
+    uart_fifo_fill(dev_upload, (const uint8_t *)buf, len);
```

Comment:
This is how all the Zephyr examples use it, my understanding is it means I don't want to do anything specific if the uart fails to copy my entire buffer.  I hate to block here and keep trying to get the uart_fifo_fill to succeed, as perhaps it could go on forever.  I'm open to suggestions however, is it expected we would keep trying until we are sure we had sent all the bytes? At what point do we give up? Perhaps @grgustaf has some insight?

Thanks for the comment.